models/repo.go

```diff
@@ -1184,26 +1184,22 @@ func CountUserRepositories(userID int64, private bool) int64 {
 }
 
 // Repositories returns all repositories
-func Repositories(opts *SearchRepoOptions) (_ []*Repository, err error) {
+func Repositories(opts *SearchRepoOptions) (_ RepositoryList, err error) {
 	if len(opts.OrderBy) == 0 {
 		opts.OrderBy = "id ASC"
 	}
 
-	repos := make([]*Repository, 0, opts.PageSize)
-	return repos, x.Limit(opts.PageSize, (opts.Page-1)*opts.PageSize).OrderBy(opts.OrderBy).Find(&repos)
-}
+	repos := make(RepositoryList, 0, opts.PageSize)
 
-// RepositoriesWithUsers returns number of repos in given page.
-func RepositoriesWithUsers(opts *SearchRepoOptions) (_ []*Repository, err error) {
-	repos, err := Repositories(opts)
-	if err != nil {
-		return nil, fmt.Errorf("Repositories: %v", err)
+	if err = x.
+		Limit(opts.PageSize, (opts.Page-1)*opts.PageSize).
+		OrderBy(opts.OrderBy).
+		Find(&repos); err != nil {
+		return nil, fmt.Errorf("Repo: %v", err)
 	}
 
-	for i := range repos {
-		if err = repos[i].GetOwner(); err != nil {
-			return nil, err
-		}
+	if err = repos.loadAttributes(x); err != nil {
```

Comment:
Yes. Ony two condition need to `loadAttributes`?

1. `Searcher` is not nil.
2. Star page of user profile

https://github.com/go-gitea/gitea/pull/940/commits/a8edb4b1cddb815c33dfd769b2c7fdb8a179aec8